system-helper/flatpak-system-helper.c

```diff
@@ -678,6 +678,61 @@ handle_configure_remote (FlatpakSystemHelper *object,
 }
 
 static gboolean
+handle_configure (FlatpakSystemHelper *object,
+		  GDBusMethodInvocation *invocation,
+		  guint arg_flags,
+		  const gchar *arg_key,
+		  const gchar *arg_value,
+		  const gchar *arg_installation)
+{
+  g_autoptr(FlatpakDir) system = NULL;
+  g_autoptr(GError) error = NULL;
+
+  g_debug ("Configure %u %s=%s %s", arg_flags, arg_key, arg_value, arg_installation);
+
+  system = dir_get_system (arg_installation, &error);
+  if (system == NULL)
+    {
+      g_dbus_method_invocation_return_gerror (invocation, error);
+      return TRUE;
+    }
+
+  if ((arg_flags & ~FLATPAK_HELPER_CONFIGURE_FLAGS_ALL) != 0)
+    {
+      g_dbus_method_invocation_return_error (invocation, G_DBUS_ERROR, G_DBUS_ERROR_INVALID_ARGS,
+                                             "Unsupported flags enabled: 0x%x", (arg_flags & ~FLATPAK_HELPER_CONFIGURE_FLAGS_ALL));
+      return TRUE;
+    }
+
+  /* We only support this for now */
+  if (strcmp (arg_key, "languages") != 0)
```

Comment:
Is it worth also validating (either here, or in repo_set_config) that the set value conforms to some idea of what the syntax is meant to be? Or you think the downside to a garbage subpath (ie, no locales get installed) isn't bad enough to de-genericise the codepaths?

Reply:
Its hard to limit the set to supported locales, since that list is not static. But, we could warn about e.g. weird characters maybe.

I dunno, it doesn't strike me as super important, and may be tricky for other config options we add later.